firmware/lib/screenManager/screenManager.h

```diff
@@ -5,29 +5,86 @@
 #include <config_helper.h>
 #include <TFT_eSPI.h>
 #include <SPI.h>
+#include <OpenFontRender.h>
+#include "../../include/ttffonts.h"
 
 #define NUM_SCREENS 5
 
-// Define your class or functions here
+#ifndef DEFAULT_FONT
+#define DEFAULT_FONT ROBOTO_REGULAR
+#endif
+
+// All available TTF fonts
+enum TTF_Font {
+    NONE,
+    ROBOTO_REGULAR,
+    FINAL_FRONTIER,
+    DSEG7,
+    DSEG14
+};
+
+struct TTF_FontScale {
+    TTF_Font font;
+    float scale;
+};
+
+// Set scaling factor for each font if necessary
+const TTF_FontScale fontScaleFactors[] = { { ROBOTO_REGULAR, 1.37 }, { FINAL_FRONTIER, 1.5 } };
 
 class ScreenManager {
 public:
     ScreenManager(TFT_eSPI& tft);
 
+    void setFont(TTF_Font font);
     TFT_eSPI& getDisplay();
+    OpenFontRender &getRender();
 
     void selectScreen(int screen);
     void selectAllScreens();
     void reset();
 
     void fillAllScreens(uint32_t color);
     void clearAllScreens();
+    void fillScreen(uint32_t color);
+    void clearScreen(int screen=-1);
+
+    // Set TTF parameters for next drawString()
+    void setFontColor(uint32_t color);
+    void setFontColor(uint32_t color, uint32_t background);
+    void setBackgroundColor(uint32_t color);
+    void setFontSize(uint32_t size);
+    void setAlignment(Align align);
 
-    void clearScreen(int screen);
+    // Helper functions
+    unsigned int calculateFitFontSize(uint32_t limit_width, uint32_t limit_height, Layout layout, const char *str);
+
+    // Draw string functions
+    void drawString(const char *text, int x, int y, unsigned int fontSize, Align align, u_int16_t fgColor, uint16_t bgColor, bool applyScale=true);
+    void drawString(const char *text, int x, int y, unsigned int fontSize, Align align, u_int16_t fgColor);
+    void drawString(const char *text, int x, int y, unsigned int fontSize, Align align);
+    void drawString(const char *text, int x, int y);
```

Comment:
I don't think these (and other drawing functions) belong in the screenManager class?
Probably a `Renderer` or `Graphics` class.

Reply:
Well, I see your point, but I also see the convenience this offers.
See also my main comment below.